### PR TITLE
Develop owls 84334

### PR DESCRIPTION
**Pull request by qvdang**
This PR updates the domain introspection script to obtain the administration port in the context of the domain's secure mode,
especially when secure mode is turned on by default(planned for 14.1.2.0).

Test results
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1789/
Test Result (3 failures / +3)
    oracle.weblogic.kubernetes.ItConfigDistributionStrategy.oracle.weblogic.kubernetes.ItConfigDistributionStrategy
    oracle.weblogic.kubernetes.ItCoherenceTests.testCohernceServerRollingRestart
    oracle.weblogic.kubernetes.ItOperatorRestart.testOperatorRestartWhenPodRoll

Ran each of these failed tests individually and they all passed:
* https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1793/testReport/oracle.weblogic.kubernetes/ItConfigDistributionStrategy/
* https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1794/testReport/oracle.weblogic.kubernetes/ItCoherenceTests/
* https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1799/testReport/oracle.weblogic.kubernetes/ItOperatorRestart/

### CODE REVIEW

File: operator/src/main/resources/scripts/introspectDomain.py [hunk 1]
```diff
@@ -1664,6 +1662,50 @@ def getSSLOrNone(server):
...
+  else:
+    administrationPortEnabled = isAdministrationPortEnabledForDomain(domain)
+  return administrationPortEnabled
+
+def getAdministrationPort(server, domain):
+  port = server.getAdministrationPort()
+  # In off-line WLST, the server's AdministrationPort default value is 0
+  if port == 0:
+    port = domain.getAdministrationPort()
+  return port
```
Comment (tbarnes-us) [inline operator/src/main/resources/scripts/introspectDomain.py:1707]: I've found three additional attributes that this script checks, that have different default and secureValue fields, and that aren't included in this update:

- NetworkAccessPoint.Protocol: t3 --> t3s
- Server[Template].ListenPortEnabled: true --> false
- Server[Template].SSL.Enabled: false --> true

Can you update the script to handle them too?

File: operator/src/main/resources/scripts/introspectDomain.py [hunk 2]
```diff
@@ -1664,6 +1662,50 @@ def getSSLOrNone(server):
...
+    cd('ServerTemplate')
+  else:
+    cd('Server')
+  cd(server.getName())
+  if isSet('AdministrationPortEnabled'):
+    administrationPortEnabled = server.isAdministrationPortEnabled()
+  else:
+    administrationPortEnabled = isAdministrationPortEnabledForDomain(domain)
+  return administrationPortEnabled
+
```
Comment (tbarnes-us) [inline operator/src/main/resources/scripts/introspectDomain.py:1701]: I've found three additional attributes that this script checks, that have different default and secureValue fields, and that aren't included in this update:

- NetworkAccessPoint.getProtocol: t3 --> t3s (search for "getProtocol()")
- Server[Template].isListenPortEnabled: true --> false (search for "isListenPortEnabled()")
- SSL.isEnabled: false --> true (search for "isEnabled()")

I assume the script needs to be updated for them too?

Comment (tbarnes-us) [reply operator/src/main/resources/scripts/introspectDomain.py:1701]: (The current pull looks excellent to me otherwise BTW.)

Comment (qvdang) [reply operator/src/main/resources/scripts/introspectDomain.py:1701]: Good catch.  Let me add some updates for those as well. Thanks!

File: operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java [hunk 3]
```diff
@@ -535,7 +551,9 @@ protected V1ObjectMeta createMetadata() {
           .ifPresent(hash -> addHashLabel(metadata, LabelConstants.MODEL_IN_IMAGE_MODEL_SECRETS_HASH, hash));
 
     // Add prometheus annotations. This will overwrite any custom annotations with same name.
-    AnnotationHelper.annotateForPrometheus(metadata, getDefaultPort());
+    // Prometheus does not support "prometheus.io/scheme".  The scheme(http/https) can be set
+    // in the Prometheus Chart values yaml under the "extraScrapeConfigs:" section.
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java:555]: Do we need to make changes elsewhere to accommodate these `extraScrapeConfigs`? For instance, are there documentation or blogs that must now be updated?

**Comment by rjeberhard** [comment]
@qvdang and @mriccell, this PR looks good to me. The only question (I commented above) is on the code comment and change around the Prometheus annotation. I understand the comment, but it seems to imply that we have additional doc. or blog work elsewhere. Is that true?

**Review by rjeberhard — APPROVED** [review]

File: operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java (hunk 3 above)
Comment (qvdang) [reply operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java:555]: @rjeberhard , I think we do for the use case when only the SSL port of the managed server is enabled.  The other use case I found that breaks the monitoring exporter app is when the managed server has the non-SSL port enabled + the administration port enabled.  Should I enter a tracking bug or JIRA for it?

File: operator/src/main/resources/scripts/introspectDomain.py (hunk 2 above)
Comment (tbarnes-us) [reply operator/src/main/resources/scripts/introspectDomain.py:1729]: Looks good!

**Review by tbarnes-us — APPROVED** [review]
LGTM

**Comment by qvdang** [comment]
I.T. results:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2173
Test Result (6 failures / +1)
    oracle.weblogic.kubernetes.ItMiiSample.testJrfUpdate3UseCase
    oracle.weblogic.kubernetes.ItServerStartPolicy.testDomainRestart
    oracle.weblogic.kubernetes.ItServerStartPolicy.testDynamicClusterRestart
    oracle.weblogic.kubernetes.ItServerStartPolicy.testAdminServerRestart
    oracle.weblogic.kubernetes.ItServerStartPolicy.testConfigClusterStartServerUsingAlways
    oracle.weblogic.kubernetes.ItServerStartPolicy.testDynamicClusterStartServerUsingAlways

Reruns of failed tests(both were clean):
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2211/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2204/

**Pull request merged**